src/firebase-app.ts

```diff
@@ -428,51 +422,24 @@ export class FirebaseApp implements app.App {
     this.INTERNAL.delete();
 
     return Promise.all(Object.keys(this.services_).map((serviceName) => {
-      return this.services_[serviceName].INTERNAL.delete();
+      const service = this.services_[serviceName];
+      if (isStateful(service)) {
+        return service.delete();
+      }
+      return Promise.resolve();
     })).then(() => {
       this.services_ = {};
       this.isDeleted_ = true;
     });
   }
 
-  private ensureService_<T extends FirebaseServiceInterface>(serviceName: string, initializer: () => T): T {
-    this.checkDestroyed_();
-
-    let service: T;
-    if (serviceName in this.services_) {
-      service = this.services_[serviceName] as T;
-    } else {
-      service = initializer();
-      this.services_[serviceName] = service;
-    }
-    return service;
-  }
-
-  /**
-   * Returns the service instance associated with this FirebaseApp instance (creating it on demand
-   * if needed). This is used for looking up monkeypatched service instances.
-   *
-   * @param serviceName The name of the service instance to return.
-   * @return The service instance with the provided name.
-   */
-  private getService_(serviceName: string): FirebaseServiceInterface {
+  private ensureService_<T>(serviceName: string, initializer: () => T): T {
     this.checkDestroyed_();
-
     if (!(serviceName in this.services_)) {
-      this.services_[serviceName] = this.firebaseInternals_.serviceFactories[serviceName](
-        this,
-        this.extendApp_.bind(this),
-      );
+      this.services_[serviceName] = initializer();
     }
 
-    return this.services_[serviceName];
-  }
-
-  /**
-   * Callback function used to extend an App instance at the time of service instance creation.
-   */
-  private extendApp_(props: {[prop: string]: any}): void {
-    deepExtend(this, props);
+    return this.services_[serviceName] as T;
```

Comment:
After you removed the tests in `describe('#[service]()', () => {...}` below, do you still have tests to confirm this, eg. the cached service is returned?

Reply:
The removed tests were only verifying the caching of monkey-patched services. We don't have any monkey-patched services in the SDK. There are some tests in `firebase-app.spec.ts` to test for caching behavior of actual services like Auth. I've now added some new tests to `firebase-namespace.spec.ts` to verify the same at the FirebaseNamespace level.